### PR TITLE
Resolve #60 Seq2Seq Example with GPU Support

### LINKED ISSUE #60: Seq2Seq Example with GPU Support
Hello,

How can I run the Seq2Seq example with my GPU?

I already modified the training data to use the cuda device as well as the model:
```
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu") 

train_data = tx.data.PairedTextData(hparams=config_data.train, device=device)
val_data = tx.data.PairedTextData(hparams=config_data.val, device=device)
test_data = tx.data.PairedTextData(hparams=config_data.test, device=device)

model = Seq2SeqAttn(train_data)
model.to(device)
```

### PR DESCRIPTION
**Pull request by gpengzhi**
#60

### CODE REVIEW

File: examples/seq2seq_attn/seq2seq_attn.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def forward(self, batch, mode):
                 decoding_strategy="infer_greedy",
                 embedding=self.target_embedder,
                 start_tokens=start_tokens,
-                end_token=torch.tensor(self.eos_token_id))
+                end_token=start_tokens.new_tensor(self.eos_token_id))
```
Comment (huzecong) [inline examples/seq2seq_attn/seq2seq_attn.py]: Let's change this to a Python int, per #63.

File: examples/seq2seq_attn/seq2seq_attn.py [hunk 2]
```diff
@@ -100,11 +100,12 @@ def forward(self, batch, mode):
         else:
             start_tokens = torch.ones_like(batch['target_length']) * \
                            self.bos_token_id
+
             helper_infer = self.decoder.create_helper(
                 decoding_strategy="infer_greedy",
                 embedding=self.target_embedder,
                 start_tokens=start_tokens,
-                end_token=start_tokens.new_tensor(self.eos_token_id))
```
Comment (huzecong) [inline examples/seq2seq_attn/seq2seq_attn.py]: I might be wrong, but isn't `self.eos_token_id` already a Python int? Could you test run this to make sure it works?

Comment (gpengzhi) [reply examples/seq2seq_attn/seq2seq_attn.py]: `self.eos_token_id` is a `numpy.int64`. If we use `self.eos_token_id` directly, we will get an error here (https://github.com/asyml/texar-pytorch/blob/master/texar/modules/decoders/decoder_helpers.py#L247).

Comment (huzecong) [reply examples/seq2seq_attn/seq2seq_attn.py]: I see, thanks for clarifying!

**Pull request merged**